### PR TITLE
Refactoring and code improving for OddEvenSort

### PR DESCRIPTION
**Pull request by alxkm**
Refactoring and code improving for OddEvenSort, changing it according to SortAlgorithm approach, also applying SortUtils

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/sorts/OddEvenSort.java [hunk 1]
```diff
@@ -2,68 +2,62 @@
...
+        }
+        System.out.println("--------------");
+        final OddEvenSort oddEvenSort = new OddEvenSort();
+        oddEvenSort.sort(arr);
+
+        // Print Sorted elements
+        for (int i = 0; i < arr.length - 1; ++i) {
+            System.out.println(arr[i]);
+            assert arr[i] <= arr[i + 1];
+        }
```
Comment (vil02) [inline src/main/java/com/thealgorithms/sorts/OddEvenSort.java:61]: The `main` method is completely redundant here.

```suggestion
```

File: src/test/java/com/thealgorithms/sorts/OddEvenSortTest.java [hunk 2]
```diff
@@ -9,28 +9,34 @@
...
     @Test
     public void oddEvenSortIntegerArray() {
-        int[] inputArray = {57, 69, -45, 12, -85, 3, -76, 36, 67, -14};
-        OddEvenSort.oddEvenSort(inputArray);
-        int[] expectedOutput = {-85, -76, -45, -14, 3, 12, 36, 57, 67, 69};
+        Integer[] inputArray = {57, 69, -45, 12, -85, 3, -76, 36, 67, -14};
+        oddEvenSort.sort(inputArray);
+        Integer[] expectedOutput = {-85, -76, -45, -14, 3, 12, 36, 57, 67, 69};
         assertArrayEquals(inputArray, expectedOutput);
     }
```
Comment (vil02) [inline src/test/java/com/thealgorithms/sorts/OddEvenSortTest.java:41]: I think one can remove these test cases - they are covered in `SortingAlgorithmTest`.

File: src/main/java/com/thealgorithms/sorts/OddEvenSort.java [hunk 3]
```diff
@@ -2,68 +2,62 @@
 
 import java.util.Random;
```
Comment (vil02) [inline src/main/java/com/thealgorithms/sorts/OddEvenSort.java:3]: This is not needed, when `main` is removed.

```suggestion
```

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/sorts/OddEvenSort.java (hunk 1 above)
Comment (alxkm) [reply src/main/java/com/thealgorithms/sorts/OddEvenSort.java:61]: Agree, this was before the recent changes, so I didn't know. We can either delete it or leave it as is. It's clear now, thank you.

**Review by vil02 — APPROVED** [review]

**Pull request merged**